### PR TITLE
[pytorch][build] Added apt upgrades to Pt1.9-cpu,gpu,inf,train and updated Pillow to version 8.3.1

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Adding APT Upgrade helps us consume security patches while rebuilding images and thus pass the sanity tests.

### DLC image/dockerfile
1. deep-learning-containers/pytorch/training/docker/1.9/py3/cu111/Dockerfile.gpu
2. deep-learning-containers/pytorch/inference/docker/1.9/py3/Dockerfile.cpu
3. deep-learning-containers/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu
4. deep-learning-containers/pytorch/training/docker/1.9/py3/Dockerfile.cpu

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
Add skips if needed, get PR merge ready, and we should be good

File: pytorch/training/docker/1.9/py3/cu111/Dockerfile.gpu [hunk 1]
```diff
@@ -47,12 +47,13 @@ ENV MANUAL_BUILD=0
...
 
 ARG PT_TRAINING_URL=https://aws-pytorch-cicd-v3-binaries.s3.us-west-2.amazonaws.com/r1.9.0_aws_v3/20210621-225848/ba7c4b277d2664f51afae8f0bfef33ec82dd6018/gpu/torch-1.9.0-cp38-cp38-manylinux1_x86_64.whl
-ARG PT_TORCHVISION_URL=https://torchvision-build-rc.s3.us-west-2.amazonaws.com/1.9.0/gpu/torchvision-0.10.0-cp38-cp38-linux_x86_64.whl
+ARG PT_TORCHVISION_URL=https://torchvision-build-rc.s3.us-west-2.amazonaws.com/1.9.0/gpu/07-30-2021/torchvision-0.10.0-cp38-cp38-linux_x86_64.whl
 ARG SMD_MODEL_PARALLEL_URL=https://sagemaker-distributed-model-parallel.s3.us-west-2.amazonaws.com/pytorch-1.9.0/build-artifacts/2021-06-29-16-48/smdistributed_modelparallel-1.4.0-cp38-cp38-linux_x86_64.whl
 ARG SMDATAPARALLEL_BINARY=https://smdataparallel.s3.amazonaws.com/binary/pytorch/1.9.0/cu111/2021-06-21/smdistributed_dataparallel-1.2.1-cp38-cp38-linux_x86_64.whl
 ARG PT_S3_WHL_GPU=https://aws-s3-plugin.s3.us-west-2.amazonaws.com/binaries/0.0.1/1c3e69e/awsio-0.0.1-cp38-cp38-manylinux1_x86_64.whl
 
 RUN apt-get update \
+ && apt-get -y upgrade --only-upgrade systemd \
```
Comment (arjkesh) [inline pytorch/training/docker/1.9/py3/cu111/Dockerfile.gpu:56]: Add comment
```
# TODO: Remove systemd upgrade once it is updated in base image
```

File: pytorch/training/docker/1.9/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -29,12 +29,13 @@ ENV SAGEMAKER_TRAINING_MODULE=sagemaker_pytorch_container.training:main
...
 
 ARG PT_TRAINING_URL=https://aws-pytorch-cicd-v3-binaries.s3.us-west-2.amazonaws.com/r1.9.0_aws_v3/20210621-225848/ba7c4b277d2664f51afae8f0bfef33ec82dd6018/cpu/torch-1.9.0-cp38-cp38-manylinux1_x86_64.whl
-ARG PT_TORCHVISION_URL=https://torchvision-build-rc.s3.us-west-2.amazonaws.com/1.9.0/cpu/torchvision-0.10.0-cp38-cp38-linux_x86_64.whl
+ARG PT_TORCHVISION_URL=https://torchvision-build-rc.s3.us-west-2.amazonaws.com/1.9.0/cpu/07-30-2021/torchvision-0.10.0-cp38-cp38-linux_x86_64.whl
 ARG PT_S3_WHL=https://aws-s3-plugin.s3.us-west-2.amazonaws.com/binaries/0.0.1/1c3e69e/awsio-0.0.1-cp38-cp38-manylinux1_x86_64.whl
 
 WORKDIR /
 
 RUN apt-get update \
+ && apt-get -y upgrade --only-upgrade systemd \
```
Comment (arjkesh) [inline pytorch/training/docker/1.9/py3/Dockerfile.cpu:38]: Add comment
```
# TODO: Remove systemd upgrade once it is updated in base image
```

File: pytorch/inference/docker/1.9/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -18,8 +18,11 @@ ENV LD_LIBRARY_PATH /opt/conda/lib/:$LD_LIBRARY_PATH
 ENV PATH /opt/conda/bin:$PATH
 ENV SAGEMAKER_SERVING_MODULE sagemaker_pytorch_serving_container.serving:main
 ENV TEMP=/home/model-server/tmp
+# Set MKL_THREADING_LAYER=GNU to prevent issues between torch and numpy/mkl
+ENV MKL_THREADING_LAYER=GNU
 
 RUN apt-get update \
+ && apt-get -y upgrade --only-upgrade systemd \
```
Comment (arjkesh) [inline pytorch/inference/docker/1.9/py3/Dockerfile.cpu:25]: Add comment
```
# TODO: Remove systemd upgrade once it is updated in base image
```

File: pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu [hunk 4]
```diff
@@ -30,8 +30,11 @@ ENV EFA_VERSION=1.11.2
 ENV OMPI_VERSION=4.1.1
 ENV NVML_VERSION=11.1.74
 ENV TORCH_CUDA_ARCH_LIST="3.7 5.0 7.0+PTX 8.0"
+# Set MKL_THREADING_LAYER=GNU to prevent issues between torch and numpy/mkl
+ENV MKL_THREADING_LAYER=GNU
 
 RUN apt-get update \
+    && apt-get -y upgrade --only-upgrade systemd \
```
Comment (arjkesh) [inline pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:37]: Add comment
```
# TODO: Remove systemd upgrade once it is updated in base image
```

File: test/dlc_tests/sanity/test_safety_check.py [hunk 5]
```diff
@@ -148,13 +148,13 @@ def _get_latest_package_version(package):
...
 @pytest.mark.canary("Run safety tests regularly on production images")
 @pytest.mark.skipif(not is_dlc_cicd_context(), reason="Skipping test because it is not running in dlc cicd infra")
-@pytest.mark.skipif(
-    not (is_mainline_context() or (is_canary_context() and is_time_for_canary_safety_scan())),
-    reason=(
-        "Skipping the test to decrease the number of calls to the Safety Check DB. "
-        "Test will be executed in the 'mainline' pipeline and canaries pipeline."
-    )
-)
+# @pytest.mark.skipif(
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_safety_check.py:151]: revert temp changes

File: dlc_developer_config.toml [hunk 6]
```diff
@@ -13,17 +13,17 @@ benchmark_mode = false
 
 [build]
 # Frameworks for which you want to disable both builds and tests
-skip_frameworks = []
```
Comment (arjkesh) [inline dlc_developer_config.toml:16]: revert temp changes

**Review by arjkesh — APPROVED** [review]
